### PR TITLE
refactor: Use method references in lambda

### PR DESCRIPTION
**Pull request by timtebeek**
Use this link to re-run the recipe: https://app.moderne.io/recipes/org.openrewrite.staticanalysis.ReplaceLambdaWithMethodReference?organizationId=T3BlblJld3JpdGU%3D

### CODE REVIEW

File: src/main/java/org/openrewrite/analysis/dataflow/DataFlowNode.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public static Option<DataFlowNode> of(Cursor cursor) {
                     .toOption()
                     .map(expr -> new ExpressionDataFlowNode(cursor, expr));
         } else if (cursor.getValue() instanceof J.VariableDeclarations.NamedVariable) {
-            return Parameter.viewOf(cursor).map(parameter -> (DataFlowNode) new ParameterDataFlowNode(cursor, parameter)).toOption();
+            return Parameter.viewOf(cursor).map(DataFlowNode.class::cast).toOption();
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/analysis/dataflow/DataFlowNode.java:68]: Well this is a bug. :/

Comment (timtebeek) [inline src/main/java/org/openrewrite/analysis/dataflow/DataFlowNode.java:68]:
```suggestion
            return Parameter.viewOf(cursor).map(parameter -> (DataFlowNode) new ParameterDataFlowNode(cursor, parameter)).toOption();
```

File: src/main/java/org/openrewrite/analysis/dataflow/global/GlobalDataFlowAccumulator.java [hunk 2]
```diff
@@ -346,7 +346,7 @@ public GlobalDataFlow.Summary summary(Cursor cursor) {
         }
         return DataFlowNode
                 .of(cursor)
-                .map(n -> (GlobalDataFlow.Summary) new ResultSummary(n, prunedParticipatingNodes))
+                .map(org.openrewrite.analysis.dataflow.global.GlobalDataFlow.Summary.class::cast)
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/analysis/dataflow/global/GlobalDataFlowAccumulator.java:349]:
```suggestion
                .map(n -> (GlobalDataFlow.Summary) new ResultSummary(n, prunedParticipatingNodes))
```

File: src/main/java/org/openrewrite/analysis/dataflow/DataFlowNode.java (hunk 1 above)
Comment (timtebeek) [reply src/main/java/org/openrewrite/analysis/dataflow/DataFlowNode.java]: Logged separately in https://github.com/openrewrite/rewrite-static-analysis/issues/150

File: src/main/java/org/openrewrite/analysis/controlflow/ControlFlowSummary.java [hunk 3]
```diff
@@ -59,16 +59,16 @@ private static void recurseGetAllControlFlowNodes(ControlFlowNode current, Set<C
     public Set<ControlFlowNode.BasicBlock> getBasicBlocks() {
         return getAllNodes()
                 .stream()
-                .filter(node -> node instanceof ControlFlowNode.BasicBlock)
-                .map(node -> (ControlFlowNode.BasicBlock) node)
+                .filter(org.openrewrite.analysis.controlflow.ControlFlowNode.BasicBlock.class::isInstance)
```
Comment (JLLeitschuh) [inline src/main/java/org/openrewrite/analysis/controlflow/ControlFlowSummary.java:62]: Why use this FQN. Please don't use the FQN when it isn't necessary

**Review by JLLeitschuh — COMMENTED** [review]
Please resolve the identified issues

**Pull request merged**